Update Navbar and Hero to be consistent from page to page

mikenicklas opened the pull request:
The new landing page design and the old design are very different aesthetically. This carries the risk of "frankenstyling" the marketing site because customers have two very different experiences between the homepage, the docs page, handbook, etc.

One way to minimize this is to make some of the components universal from page-to-page. The most obvious ones we can share are the header and footer.

This PR adds a universal header (with dark mode support) which will be rendered on every page. The only difference between the homepage header styling and the "docs" styling is the width of the header container. With time, the layout of the docs section will match the landing page.
![pr-universal-header](https://user-images.githubusercontent.com/6363580/109010152-71f80000-767d-11eb-9e76-ecf1ca2aa88a.gif)

For now, we are prioritizing time so menu items are hardcoded and we haven't extracted every possible component. After we launch the new landing page, we will continue to extract components and wire up various graphql queries.

Please review the [branch preview](https://deploy-preview-1019--posthog.netlify.app/) and let me know if you spot anything concerning!

corywatilo commented:
Looking great! 💪🏼 I like what you did on the features page to use the header background color in the hero section.

## Homepage

**Content**

- [ ] Since Community just links to Github right now, let's just label it "GitHub" until we have our whole dropdown menu in there

**Design**

- [ ] Buttons: Use `Gosha Sans`, `15px`, remove `letter-spacing: .1em` (should apply to both primary and secondary)
- [ ] `h1` line-height: reduce to around `60px` (currently at `72px`)
- [ ] _"Join 2,700 companies"_ - reduce font size a bit so it fits on one line on a normal desktop browser
- [ ] Footer links: Can we add a little more padding between columns?
- [ ] Footer link column headers (`.footer-links-header`) - increase size so there's a little more contrast in size between headers and links

## Other pages

- [ ] `h1` size on homepage is okay, but feels too large on inner pages. Can we use a slightly smaller `h1` font for those?
- [ ] Actually in general, [all font sizes](https://www.dropbox.com/s/40si2eg3cewo7ib/Screen%20Shot%202021-02-24%20at%209.54.32%20AM.png?dl=0) feel a little large. Sidebar links and body font could go from `18px` -> `16px`, and all other header sizes can come down a bit more

### Handbook/Docs

In **dark mode** on [these sections](https://deploy-preview-1019--posthog.netlify.app/handbook/company/story), we have a deep blue header and a dark gray body background. Using Inspector, I made [a few tweaks](https://www.dropbox.com/s/k8fxp7vdux9mj4t/Screen%20Shot%202021-02-24%20at%209.39.24%20AM.png?dl=0) to dark mode which would make them feel a little more cohesive:

1. Use header background color in body: `#220f3f`
2. Sidebar background: `#19082F`
3. For sidebar, I added some small styling: border radius, left margin (but we may have issues when switching back to light mode)
4. Not reflected in my screenshot, but jump links on right side (`.ant-anchor-link`) need to be white with some transparency (around `rgba(255,255,255,.4)`) vs current `#4e4c4c` (this is for `body.dark .post-page-wrapper .ant-anchor-link a`)

- [ ] Body text font [feels too heavy](https://www.dropbox.com/s/m2atldny8jwgva8/Screen%20Shot%202021-02-24%20at%209.47.46%20AM.png?dl=0). (It looks a lot thinner in the mocks, using `Medium` - is that what the text is using?)
- [ ] Let's fix up [bullet points](https://www.dropbox.com/s/py77ywm0g0lcbyr/Screen%20Shot%202021-02-24%20at%209.45.55%20AM.png?dl=0) (have bullets line up with text, match paragraph font size, line height)

### Pricing

- [ ] Increase line height between [plan line items](https://www.dropbox.com/s/6fah6tghfcwe4jk/Screen%20Shot%202021-02-24%20at%209.50.00%20AM.png?dl=0)

### Features

- [ ] Reduce [tagline line height](https://www.dropbox.com/s/6miokgqz4nu9nhj/Screen%20Shot%202021-02-24%20at%209.51.08%20AM.png?dl=0) (currently 40px, 30px feels about right)

yakkomajuri commented on src/components/Header/Header.tsx:
```diff
@@ -0,0 +1,161 @@
+    )
+}
+
+export default Header
```
While we're refactoring JS -> TS, can we also move away from the default export?

Here's Marius making his case to me a while ago:
https://github.com/PostHog/posthog.com/pull/511#discussion_r503230523

yakkomajuri reviewed (changes requested):
Great stuff! A few minor things here and there but otherwise looks good.

Thanks for the JS -> TS move btw

yakkomajuri commented:
Another thing on the UI side, the initial render is now not as nice, with some jumpiness in the hero

https://user-images.githubusercontent.com/38760734/109021170-8d313280-76b2-11eb-99c7-c437d43804e9.mov

mikenicklas commented on src/components/Header/Header.tsx:
```diff
@@ -0,0 +1,161 @@
+            </button>
+        </li>
+    )
+}
```
@yakkomajuri I think I'd like to leave this one here for now. It's just a placeholder until we work through the landing page where we'll have a separate button components that we should be using globally. This one was just intended for the CTA in the navbar but because it had to duplicate between horizontal nav, and collapsed mobile nav I figured I'd extract it in the same file so the class names weren't duplicated and likely to get out of sync.

Is that an okay path for you?

yakkomajuri replied:
Fine by me! This was more of a nit than anything

mikenicklas commented:
@yakkomajuri @corywatilo I made a lot of the revisions suggested here. A couple of notes on the few that were left out:

- Dark mode/Kea -> For now, I think we should we go ahead and skip this. Continuing down the current path of overrides will make maintenance tricky and we can/should definitely circle back to this. I'm going to create a specific issue to fine-tune how we are doing Dark mode and hook it in with Tailwind's dark mode support. We should strive to have a seamless dark mode that *never* shows the incorrect theme but I don't think right now is the time to get into the weeds with that.
- Some smaller styling details in the Docs/Handbook -> @corywatilo I made the vast majority of the changes you suggested but there were a few that would require a lot of hacking to fix. CSS is layered between Antd/Page stylesheets/Component stylesheets/styled components and now Tailwind. As we move our way through the marketing site, we can begin to delete old styles & eventually kill off Ant as we move towards our own custom components styled via Tailwind.

If you see anything you'd consider absolutely blocking, please feel free to suggest changes here. Otherwise, I think we should stay focused on the plan that while we're going through a full redesign there may be some small details that aren't great. Maybe we can open a long running issue so in the short term we can log these inconveniences and make sure to address them as we work through the different sections of the site.

corywatilo commented:
This looks great! My only nit is that the "Login" text in the header is 1px-2px too big. If we can just tweak that, I don't see why else why we can't get this out there! 💪🏼

jamesefhawkins commented:
The only blocker I can see is the internal blog pages are sad:

<img width="1569" alt="Screenshot 2021-03-01 at 15 22 26" src="https://user-images.githubusercontent.com/47497682/109518188-ef6aa880-7aa1-11eb-882c-03a96eb6f09b.png">

mikenicklas commented:
@jamesefhawkins Good catch. Looks like some weird leftover absolute positioning. Just did a quick pass to make it usable and then we can reskin it as we move further into the project.

@corywatilo Updated the login text size 🙌

yakkomajuri commented on src/components/Header/Header.tsx:
```diff
@@ -0,0 +1,156 @@
+
+    const themeSupportedColor = websiteTheme === 'light' ? 'bg-lightmode-gray' : 'bg-darkmode-purple'
+
+    const darkModeSupportedPage = isDocsPage || onPostPage
```
all docs pages are post pages

corywatilo commented:
Can we add font-smoothing: antialiased to devices with hi-dpi displays? I think around 1.5 or 1.3 pixel density does the trick IIRC.

kpthatsme commented:
Can we modify the hero and add a tab to represent Session recordings in the Hero image? 

I think that's one of the features that draws a set of folks in and definitely helps reinforce the 'complete stack' message.

Twixes commented:
Solid step forward. I only noticed 2 regressions:

- [ ] Hero space is a bit jumpy on load in all viewport widths, but in the "CTA left of illustration" layout especially so, with an unexpected animation.
    ![Load](https://user-images.githubusercontent.com/4550621/110011823-24842e80-7d20-11eb-85df-f1f5c3c85752.gif)

- [ ] The "H" in "GitHub" should be capitalized
    ![Github](https://user-images.githubusercontent.com/4550621/110011423-b8092f80-7d1f-11eb-922b-e13dd83374d7.png)
    As more of a UX design thing, it's a bit unexpected to me as a user that the only item to redirect away from posthog.com is in the middle of the nav. I'd put it last I think.

lottiecoxon commented:
(because my wifi is really rather bad today I can't add in photos) BUT - from the test site that was shared on slack I have one or two little issues that need ironing out. 
1. on No.5 'Improve conversion rates' - the retention image doesn't load. 
2. on No.1  'it all starts with autocapture' - the font is wrong and not gosha sans

mikenicklas commented:
@lottiecoxon @Twixes Thanks for taking a peek!

@Twixes I updated those. Good callout on the GitHub and the strange UX with it being in the middle. Fixed the casing and moved it to the end. Eventually that link will be gone and it'll swap with a community/open source section on the website directly!

@lottiecoxon 
1. Features 1-4 all use the same component so the styling should be the same. It's using Gosha for me and if you inspect the element the CSS confirms that it should be using Gosha. https://msnicklas.d.pr/ca1Xq0
2. Feature 5 "Improve conversion rates" has the image below the feature. https://msnicklas.d.pr/hSbeR8. Can you confirm if you see those?  It's a bit weird that it doesn't match the other features, but changing that is out of scope for this work.

jamesefhawkins commented:
So close. I'm seeing a big gap though on desktop (this is on a large monitor) that has suddenly appeared:

<img width="1672" alt="Screenshot 2021-03-05 at 13 19 18" src="https://user-images.githubusercontent.com/47497682/110120773-657a5280-7db5-11eb-8dec-e0639e5642ae.png">

mikenicklas commented:
@jamesefhawkins Fixed! I was trying to avoid hardcoding an arbitrary height/min-height but it looks like that's the only way to go for now.

jamesefhawkins commented:
Hooray. I've merged this, looking forward to rest of homepage!
